File: app/src/main/java/site/business/appslandz/retrobuilder/demo/MainActivity.java
package site.business.appslandz.retrobuilder.demo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import site.business.appslandz.retrobuilder.demo.NetworkInterfaces.ApiInterface;
import site.business.appslandz.retrobuilder.demo.Models.UsersList;
import site.business.appslandz.retrobuilder.Interfaces.AuthInitializer;
import site.business.appslandz.retrobuilder.Service.ApiClient;
import timber.log.Timber;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Call<UsersList> authCall = ApiClient.getInstance(this)
                .setBaseUrl("https://reqres.in")
                .getAuthClient(new AuthInitializer() {
                    @Override
                    public String getBearerToken() {
                        return "Bearer 129318309809843589350938509893540832945";
                    }
                }).create(ApiInterface.class)
                .getAllUsers();

        authCall.enqueue(new Callback<UsersList>() {
            @Override
            public void onResponse(Call<UsersList> call, Response<UsersList> response) {
                Timber.i(response.body().toString());
            }

            @Override
            public void onFailure(Call<UsersList> call, Throwable t) {
                call.cancel();
            }
        });

        Call<UsersList> openCall = ApiClient.getInstance(this)
                .setBaseUrl("https://reqres.in")
                .getClient().create(ApiInterface.class)
                .getAllUsers();


        openCall.enqueue(new Callback<UsersList>() {
            @Override
            public void onResponse(Call<UsersList> call, Response<UsersList> response) {
                Timber.i(response.body().toString());
            }

            @Override
            public void onFailure(Call<UsersList> call, Throwable t) {
                call.cancel();
            }
        });


    }
}

File: app/src/main/java/site/business/appslandz/retrobuilder/demo/MainActivity.kt
package site.business.appslandz.retrobuilder.demo

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import site.business.appslandz.retrobuilder.Interfaces.AuthInitializer
import site.business.appslandz.retrobuilder.Service.ApiClient
import site.business.appslandz.retrobuilder.Service.ApiResponse
import site.business.appslandz.retrobuilder.demo.Models.User
import site.business.appslandz.retrobuilder.demo.Models.UsersList
import site.business.appslandz.retrobuilder.demo.NetworkInterfaces.ApiInterface
import timber.log.Timber

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ApiClient.getInstance(this)
                .setBaseUrl("https://reqres.in")

        val authCall = ApiClient.getInstance(this)
                .getAuthClient(object : AuthInitializer {
                    override val bearerToken: String
                        get() = "Bearer 129318309809843589350938509893540832945"
                }).create(ApiInterface::class.java).getUsers()
        authCall.enqueue(object : Callback<ApiResponse<List<User>>> {
            override fun onFailure(call: Call<ApiResponse<List<User>>>, t: Throwable) {
                call.cancel()
                t.printStackTrace()
            }

            override fun onResponse(call: Call<ApiResponse<List<User>>>, response: Response<ApiResponse<List<User>>>) {
                Timber.d(response.body()?.data.toString())
            }

        })
    }
}